packages/compass/src/app/theme.ts

```diff
@@ -0,0 +1,32 @@
+import type { THEMES } from 'compass-preferences-model';
+import preferences from 'compass-preferences-model';
+import * as darkreader from 'darkreader';
+import * as remote from '@electron/remote';
+
+const darkreaderOptions = { brightness: 100, contrast: 90, sepia: 10 };
+
+const compassThemeToElectronTheme: Record<THEMES, typeof remote.nativeTheme.themeSource> = {
+  DARK: 'dark',
+  LIGHT: 'light',
+  OS_THEME: 'system'
+} as const;
+
+function onNativeThemeUpdated () {
+  if (!preferences.getPreferences().lgDarkmode) {
+    if (remote.nativeTheme.shouldUseDarkColors) {
+      darkreader.enable(darkreaderOptions);
+    } else {
+      darkreader.disable();
+    }
+  }
+}
+
+export function setupTheme() {
+  preferences.onPreferenceValueChanged('theme', value => {
+    const electronThemeSource = compassThemeToElectronTheme[value];
+    remote.nativeTheme.themeSource = electronThemeSource;
```

Comment:
It seems a bit more natural that this is what we should do here. Notably, this also affects other theme properties such as the window's own theme, not just its contents.